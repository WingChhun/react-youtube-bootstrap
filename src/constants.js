export const API_KEY = "AIzaSyC3ni21fos0ZbHdjOxCjXAEJBgyH4T1eo4";