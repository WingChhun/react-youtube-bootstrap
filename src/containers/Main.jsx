import React, {Component} from 'react';
import PropTypes from 'prop-types';

import {API_KEY} from "../constants";
import Searchbar from "../components/SearchBar/Searchbar";

class Main extends Component
{
    constructor(props)
    {
        super(props);
    }

    render()
    {
        return (

            <div className="Main">
                <h1>
                    Main Component</h1>

                <Searchbar/>

            </div>

        )
    }
}

export default Main;
