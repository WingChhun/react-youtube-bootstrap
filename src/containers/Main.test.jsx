import React from 'react';
import ReactDOM from 'react-dom';

import {API_KEY} from "../constants";

import Enzyme, {shallow, createMount} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Seachbar from "./Main";
import Main from './Main';

Enzyme.configure({adapter: new Adapter()});

describe("Main", () => {

    const props = {
        API_KEY: API_KEY,
        getVideos: jest.fn()
    }

    it("Renders Main without crashing", () => {

        const wrapper = shallow(<Main {...props}/>);

        expect(wrapper).toMatchSnapshot();

    })

    it("getVideos, updates state", () => {

        const wrapper = shallow(<Main {...props}/>);
        const instance = wrapper.instance();

        const videos = [
            {
                video: '1'
            }
        ]
        instance.getVideos(videos);

        expect(instance.state.videos).toBe(videos);

    });

});