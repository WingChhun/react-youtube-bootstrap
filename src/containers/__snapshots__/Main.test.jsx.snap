// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main Renders Main without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Main
    API_KEY="AIzaSyC3ni21fos0ZbHdjOxCjXAEJBgyH4T1eo4"
    getVideos={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          Main Component
</h1>,
        <Searchbar
          API_KEY="AIzaSyC3ni21fos0ZbHdjOxCjXAEJBgyH4T1eo4"
          getVideos={[Function]}
/>,
      ],
      "className": "Main",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Main Component",
        },
        "ref": null,
        "rendered": "Main Component",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "API_KEY": "AIzaSyC3ni21fos0ZbHdjOxCjXAEJBgyH4T1eo4",
          "getVideos": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            Main Component
</h1>,
          <Searchbar
            API_KEY="AIzaSyC3ni21fos0ZbHdjOxCjXAEJBgyH4T1eo4"
            getVideos={[Function]}
/>,
        ],
        "className": "Main",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Main Component",
          },
          "ref": null,
          "rendered": "Main Component",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "API_KEY": "AIzaSyC3ni21fos0ZbHdjOxCjXAEJBgyH4T1eo4",
            "getVideos": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
